### ISSUE TITLE
genome annotation fails when file of proteins is uploaded

### ISSUE THREAD

**Issue by olsonanl** [PATRIC3/patric3_website#55]
Need to add checks to ensure contigs are DNA

**Comment by dmachi** [comment]
Do you mean at the client side before submission?

**Comment by olsonanl** [comment]
well, somewhere along the line it needs to be checked. This is probably something that would go along with the auto typechecking thing (perhaps we need to distinguish between dna & protein fastas)

On Mar 17, 2015, at 2:32 PM, Dustin Machi <notifications@github.com<mailto:notifications@github.com>> wrote:


Do you mean at the client side before submission?

**Comment by mshukla1** [comment]
@olsonanl As you are making changes to the annotation service, can you add checks for proper fasta format, contig id length / format, and protein sequences?

**Comment by mshukla1** [comment]
Actually, I just submitted one annotation job with fasta protein file. It failed with the following error message. 

Using SEED id allocation from http://clearinghouse.theseed.org/Clearinghouse/clearinghouse_services.cgi
kmer_v2_data_directory = /vol/patric3/fams/2018-0531/kmers
Invalid fasta: Bad data (appears to be protein translation data) at line 2

I think this is good enough to let users know what the problem is. 

Also, I tried to upload faa file as contig file, and it did not allow me. I had to change 
it to .fasta to upload it. 

I believe these address the specific issue mentioned in the ticket. 

It would be good to have more rigorous file type checking for all file types in long run though.
